### PR TITLE
conda-build: extend matrix for python version

### PR DESCRIPTION
**Pull request by DerThorsten**
* Extend the github actions conda workflow st. we use multiple compilers and Python versions
* Check at build time if `folly>= 2023.02.13` and define a compile definition accordingly 
* run c++ tests
* skip **`StorageLock`** tests for mac as they segfault, since they are probably not supported (and they are also skipped for win)

### CODE REVIEW

**Comment by DerThorsten** [comment]
this changs the names of the workflows, so we see:
  * `linux (ubuntu-latest) Expected — Waiting for status to be reported`
  * `macos (macos-latest) Expected — Waiting for status to be reported`

File: .github/workflows/build_with_conda.yml [hunk 1]
```diff
@@ -16,7 +16,7 @@ jobs:
       matrix:
           os: [ubuntu-latest]
           python_version:  ["3.8", "3.9", "3.10"]
-          compiler: ["gcc", "clang"]
+          compiler: ["gcc", "clang", "system_compiler"]
```
[the remark this answers is not in the document]
Comment (DerThorsten) [reply .github/workflows/build_with_conda.yml:19]: we do not install this via conda:
https://github.com/man-group/ArcticDB/blob/5d1b530237b3e2e54c180a03e4bb1123475a4fdf/.github/workflows/build_with_conda.yml#L88

This is just a flag to **not** install any compiler and just use whatever is on the system.

**Comment by DerThorsten** [comment]
there sometimes seems to be segfaults on macos. But its not 100% reproducable

**Comment by DerThorsten** [comment]
so the segfaults are 100% reproducable when using `Clang` instead of `AppleClang` .
On my apple silicon I (think) I can reproduce this (might be a different segfault):

The segfault is from ` unit/arcticdb/version_store/test_filtering.py` in \ test_filter_string_equals_col_val`.
Specifically in: https://github.com/man-group/ArcticDB/blob/117811377f968957c77e8dcc108610b350ff9432/python/tests/unit/arcticdb/version_store/test_filtering.py#L42

the `read` function is segfaulting https://github.com/man-group/ArcticDB/blob/117811377f968957c77e8dcc108610b350ff9432/python/tests/unit/arcticdb/version_store/test_filtering.py#L45

The input to read is:
```
version_store=NativeVersionStore: Library: local.test_591_2023-05-31T07_56_42_310275, Primary Storage: lmdb_storage. 
symbol=test_filter_string_equals_col_val 
arctic_query=WhereClause: (Column["a"] EQ Str(vMFMBTw+BR:>wt@t[1[~R))
```

**Comment by mehertz** [comment]
@DerThorsten could we split this PR up into:

1. Matrix + C++ tests (but only testing 3.10)
2. Testing against the differing versions + compilers

?

That way we can merge the first part and have the C++ tests running for the Conda-dependency-satisfied-build whilst we work out what's going on with the other compilers?

**Pull request closed without merging**